nix/flakebox.nix

```diff
@@ -357,24 +375,29 @@ rec {
     buildPhaseCargoCommand = "patchShebangs ./scripts ; ./scripts/test/backend-test.sh";
   };
 
-  ciTestAll = craneLibTests.mkCargoDerivation {
+  ciTestAllBase = { times }: craneLibTests.mkCargoDerivation {
     pname = "${commonCliTestArgs.pname}-all";
     cargoArtifacts = workspaceBuild;
 
     # One normal run, then if succeeded, modify the "always success test" to fail,
     # and make sure we detect it (happened too many times that we didn't).
     # Thanks to early termination, this should be all very quick, as we actually
     # won't start other tests.
-    buildPhaseCargoCommand = ''
-      patchShebangs ./scripts
-      export FM_CARGO_DENY_COMPILATION=1
-      ./scripts/tests/test-ci-all.sh || exit 1
-      sed -i -e 's/exit 0/exit 1/g' scripts/tests/always-success-test.sh
-      echo "Verifying failure detection..."
-      ./scripts/tests/test-ci-all.sh 1>/dev/null 2>/dev/null && exit 1
-    '';
+    buildPhaseCargoCommand = lib.concatStringsSep "\n" (
+      lib.replicate times ''
+        patchShebangs ./scripts
+        export FM_CARGO_DENY_COMPILATION=1
+        ./scripts/tests/test-ci-all.sh || exit 1
+        sed -i -e 's/exit 0/exit 1/g' scripts/tests/always-success-test.sh
+        echo "Verifying failure detection..."
+        ./scripts/tests/test-ci-all.sh 1>/dev/null 2>/dev/null && exit 1
+      ''
+    );
```

Comment:
needs more parallelism? maybe do this in ci-test-all.sh

Reply:
I don't know if it does. At some point we'll start hitting memory limits etc. We're already waiting so long for backward compt tests in Merge Queue and nobody seems to complain, so I guess this should be fine for now too.